torch/distributed/_functional_collectives.py

```diff
@@ -226,17 +226,20 @@ def _expand_group(group: RANK_TYPES, tag: str = "") -> Tuple[str, List[int], int
         group_size = len(rankset)
         tag = tag or c10d._get_group_tag(group)
     elif isinstance(group, dt.DeviceMesh):
-        rankset = group.mesh.flatten().tolist()
-        group_size = group.mesh.size(0)
-        rankset = group.mesh.swapdims(-1, 0).reshape(-1, group_size).flatten().tolist()
-        tag = tag or c10d._get_group_tag(group.get_dim_groups()[0])
+        assert group.ndim == 1, "Only 1D mesh is supported, pass in (DeviceMesh, int) together if mesh > 1D"
+        # TODO: it should run collective in the whole mesh instead of dim 0
```

Comment:
I think tracer would just brutally trace what ever the function passed in, in this case `mesh.collective`  -> `funccol.collective`, so it would still trace this function as it, that's why we were generating all those mesh tensor CPU manipulation getattr/constants. I don't know if there's a way for `make_fx` to ignore certain piece of code for the function, so we should probably keep it tracing friendly